mysql-test/suite/rpl_nogtid/t/rpl_row_mysqlbinlog.test

```diff
@@ -144,7 +144,7 @@ connection master;
 --disable_query_log ONCE
 select "--- Test 1 Load from Dump binlog file --" as "";
 
---exec $MYSQL -e "source $MYSQLTEST_VARDIR/tmp/master.sql"
+--exec $MYSQL -c -e "source $MYSQLTEST_VARDIR/tmp/master.sql"
```

Comment:
are we fixing an unrelated test failure?

Reply:
No, this test case is related to thd->is_binlog_applier() check

Reply:
-c would send comments to server. Without this test change, how does it fail?

Reply:
Without the above mentioned condition check
DROP TABLE IF EXISTS t1,t2,t3 /* generated by server */
was getting logged

With the above mentioned condition check , we now log
DROP TABLE IF EXISTS `t1`,`t2`,`t3`

Because of this , position offsets of statements in binlog 
are different and hence test case was failing. And using -c was required to fix that.

Reply:
Does it mean,  customers/clients has to fix their client scripts to use "-c" ? :thinking:

Reply:
@nitendra-bhosle @venkatesh-prasad-v ^

Reply:
This change was required because the test does this

1. Do some transactions and record binlog position
2. Run mysqlbinlog and save the output
3. Clear the data and reset binary logs
4. Replay the binary log and verify the binlog positions

and the expectation of the test is that binlog position from step 1 should be same as step 4.

In step 1, the `DROP TABLE`  is recorded in the binlog as `DROP TABLE /* generated by server*/` But when the same binlog is replayed, it is logged as `DROP TABLE` without the comments.

So, in order to retain the comments during binlog replay, it is expected that clients should pass comments using `-c` option.

Reply:
Ok, thanks. So client doesn't need  to modify to use -c . (only the this particular test)